### PR TITLE
Dev10

### PR DESCRIPTION
**Pull request by AVassilev98**
Base spec language meets the needs for all the stub examples I've done so far

### CODE REVIEW

File: lib/seadsa/DsaLocal.cc [hunk 1]
```diff
@@ -1177,6 +1201,39 @@ void IntraBlockBuilder::visitSeaDsaFnCall(CallSite &CS) {
...
+    return;
+  }
+
+  if (isSeaDsaPtrToIntFn(callee)) {
+    // sea_dsa_ptrtoint(const void *p) -- mark the node pointed to by p as ptr to int
+    if (isSkip(*(CS.getArgument(0)))) 
+      return;
+    if (m_graph.hasCell(*(CS.getArgument(0)))) {
+      seadsa::Cell c = valueCell(*(CS.getArgument(0)));
+      c.getNode()->setRead();
```
Comment (agurfinkel) [inline lib/seadsa/DsaLocal.cc]: typo?

Comment (AVassilev98) [reply lib/seadsa/DsaLocal.cc]: Yeah, fixed in ee45aa1

File: lib/seadsa/DsaLocal.cc [hunk 2]
```diff
@@ -468,36 +479,55 @@ class IntraBlockBuilder : public InstVisitor<IntraBlockBuilder>,
...
+
+  }
+
   /// Returns true if \p F is a \p seadsa_ family of functions
   static bool isSeaDsaFn(const Function *fn) {
     if (!fn)
       return false;
     auto n = fn->getName();
-    return n.startswith("sea_dsa_") || n.startswith("seadsa_");
+    return n.startswith("sea_dsa_");
```
Comment (agurfinkel) [inline lib/seadsa/DsaLocal.cc:508]: reminder: check that tests have been updated as well

File: lib/seadsa/DsaLocal.cc [hunk 3]
```diff
@@ -468,36 +479,55 @@ class IntraBlockBuilder : public InstVisitor<IntraBlockBuilder>,
...
-    return name.equals("sea_dsa_alias") || name.equals("seadsa_alias");
-  }
-
-  static bool isSeaDsaCollapseFn(const Function *fn) {
-    if (!fn)
-      return false;
-    auto name = fn->getName();
-    // due to name change from sea_dsa to seadsa, support both version
-    return name.equals("sea_dsa_collapse") || name.equals("seadsa_collapse");
+  static bool isSeaDsaAttrbFn(const SeadsaFn &fn) {
```
Comment (agurfinkel) [inline lib/seadsa/DsaLocal.cc:511]: passing integers not by value is annoying.
I assume `SeadsaFn` is an enum, hence an int

File: lib/seadsa/DsaLocal.cc [hunk 4]
```diff
@@ -468,36 +479,55 @@ class IntraBlockBuilder : public InstVisitor<IntraBlockBuilder>,
...
   }
 
-  static bool isSeaDsaMkSequenceFn(const Function *fn) {
-    if (!fn)
-      return false;
-    auto name = fn->getName();
-    // due to name change from sea_dsa to seadsa, support both version
-    return name.equals("sea_dsa_mk_seq") || name.equals("seadsa_mk_seq");
+  auto getSeaDsaAttrbFunc(const Function *fn) {   
+    switch (getSeaDsaFn(fn)) {
```
Comment (agurfinkel) [inline lib/seadsa/DsaLocal.cc:523]: reminder: check that this works when `fn == nullptr` or change to `const Function &`

File: lib/seadsa/DsaLocal.cc [hunk 5]
```diff
@@ -1176,8 +1206,20 @@ void IntraBlockBuilder::visitIndirectCall(CallSite &CS) {
...
+      return;
+    if (m_graph.hasCell(*(CS.getArgument(0)))) {
+      seadsa::Cell c = valueCell(*(CS.getArgument(0)));
+      (c.getNode()->*getSeaDsaAttrbFunc(callee))(true);
+    }
+    return;
+  }
 
-  if (isSeaDsaAliasFn(callee)) {
+  if (fn == SeadsaFn::ALIAS) {
```
Comment (agurfinkel) [inline lib/seadsa/DsaLocal.cc:1222]: maybe a switch on `fn` instead of a block of `ifs`

File: include/seadsa/sea_dsa.h [hunk 6]
```diff
@@ -9,14 +9,20 @@ extern "C" {
   
 // sea-dsa will unify all argument's cells
 extern void seadsa_alias(const void *p, ...);
```
Comment (agurfinkel) [inline include/seadsa/sea_dsa.h:11]: unified naming convention. Probably `sea_dsa_`

**Pull request merged**